math/utils.go

```diff
@@ -21,9 +21,15 @@ func CalcEma(
 	firstTime bool,
 ) (Dec, error) {
 	// If first iteration, then return just the new value
-	if firstTime || current.Equal(previous) {
+	if firstTime || current.Equal(previous) || previous.isNaN {
+		if current.isNaN {
+			return ZeroDec(), errors.New("at least one ema operand should not be NaN")
+		}
 		return current, nil
 	}
+	if current.isNaN {
+		return previous, nil
```

Comment:
No logging here?

Reply:
yes, it could be added.
However this math package does not have any logging currently. Not sure what's the best way. 
Should we pass in a Context into the functions?

Reply:
alright i've thought further and I think that comment was too knee-jerk

we can pass the ctx in, logging is important - but i still feel that this is not the place where we need to be catching issues

it's a failure of somewhere else that we're thinking about this here

Reply:
I agree that this type of utils library is ok to have no logging, just "result or error". 
And in that light,  I think calcEMA should also fail, and not correct, current or previous EMAs. No need for logging, just fail if any piece involved in the calcs is NaN. 
I will change that

Reply:
Just pushed the change. This is also consistent with the rest of the EMA protection through the file.